src/Riok.Mapperly/Descriptors/UserMethodMappingExtractor.cs

```diff
@@ -116,11 +127,55 @@ bool isStatic
         );
     }
 
+    private static bool BuildRuntimeTargetTypeMappingParameters(
+        SimpleMappingBuilderContext ctx,
+        IMethodSymbol method,
+        out RuntimeTargetTypeMappingMethodParameters parameters
+    )
+    {
+        // reference handler parameter is always annotated
+        var refHandlerParameter = BuildReferenceHandlerParameter(ctx, method);
+        var refHandlerParameterOrdinal = refHandlerParameter?.Ordinal ?? -1;
+        var expectedParametersCount = refHandlerParameter.HasValue ? 3 : 2;
```

Comment:
Maybe extract magic numbers.

Reply:
Refactored, no need for magic numbers anymore. Code should be easier to read now.